File: codequest/2019SolutionsSG/Student Solutions/Java/Prob18.java
import java.util.Scanner;

public class Prob18 {

	public static void main(String[] args) throws Exception{
		Scanner sc = new Scanner(System.in);
		int numLoops = sc.nextInt();
		sc.nextLine();
		for(int qq = 0; qq < numLoops; qq++) {
			String aa = sc.next();
			double a = Double.parseDouble(aa);
			String bb = sc.next();
			double b = Double.parseDouble(bb);
			double aPrev = a;
			double bPrev = b;
			int numIter = 1;
			while(Math.sqrt(Math.pow(aPrev, 2) + Math.pow(bPrev, 2)) < 100) {
				double aNext = Math.pow(aPrev, 2) - Math.pow(bPrev, 2) + a;
				double bNext = 2*aPrev*bPrev + b;
				numIter++;
				aPrev = aNext;
				bPrev = bNext;
				if(numIter > 51) {
					break;
				}
			}
			String color;
			if(numIter <= 10) {
				color = "RED";
			} else if(numIter <= 20) {
				color = "ORANGE";
			} else if(numIter <= 30) {
				color = "YELLOW";
			} else if(numIter <= 40) {
				color = "GREEN";
			} else if(numIter <= 50) {
				color = "BLUE";
			} else {
				color = "BLACK";
			}
			System.out.println((aa) + "+" + bb + "i" + " " + color);
		}
		sc.close();
	}

}

File: codequest/2019SolutionsSG/Student Solutions/Python/Prob18.py
def square(a, b):
    return (a*a - b*b, 2*a*b)

def add(a1, b1, a2, b2):
    return (a1 + a2, b1 + b2)

def modulus(a, b):
    return (a*a + b*b)

for t in range(int(input())):
    A, B = input().split()
    a, b = float(A), float(B)
    a1, b1 = 0, 0
    if b == 0:
        print(A, end=' ')
    elif a == 0:
        print(B, 'i', sep='', end=' ')
    else:
        if b > 0:
            print(A, '+', B, 'i', end=' ', sep='')
        else:
            print(A, B, 'i', end=' ', sep='')
    for n in range(1, 52):
        a1, b1 = add(*square(a1, b1), a, b)
        #print(n, a1, b1, modulus(a1, b1))
        if modulus(a1, b1) > 10000:
            if n <= 10:
                print("RED")
                break
            elif n <= 20:
                print("ORANGE")
                break
            elif n <= 30:
                print("YELLOW")
                break
            elif n <= 40:
                print("GREEN")
                break
            elif n <= 50:
                print("BLUE")
                break
    else:
        print("BLACK")

    
